htlcswitch/quiescer.go

```diff
@@ -46,6 +46,51 @@ var (
 
 type StfuReq = fn.Req[fn.Unit, fn.Result[lntypes.ChannelParty]]
 
+// Quiescer is the public interface of the quiescence mechanism. Callers of the
+// quiescence API should not need any methods besides the ones detailed here.
+type Quiescer interface {
+	// IsQuiescent returns true if the state machine has been driven all the
+	// way to completion. If this returns true, processes that depend on
+	// channel quiescence may proceed.
+	IsQuiescent() bool
+
+	// QuiescenceInitiator determines which ChannelParty is the initiator of
+	// quiescence for the purposes of downstream protocols. If the channel
+	// is not currently quiescent, this method will return
+	// ErrNoDownstreamLeader.
+	QuiescenceInitiator() fn.Result[lntypes.ChannelParty]
+
+	// InitStfu instructs the quiescer that we intend to begin a quiescence
+	// negotiation where we are the initiator. We don't yet send stfu yet
+	// because we need to wait for the link to give us a valid opportunity
+	// to do so.
+	InitStfu(req StfuReq)
+
+	// RecvStfu is called when we receive an Stfu message from the remote.
+	RecvStfu(stfu lnwire.Stfu, numRemotePendingUpdates uint64) error
+
+	// CanRecvUpdates returns true if we haven't yet received an Stfu which
+	// would mark the end of the remote's ability to send updates.
+	CanRecvUpdates() bool
+
+	// CanSendUpdates returns true if we haven't yet sent an Stfu which
+	// would mark the end of our ability to send updates.
+	CanSendUpdates() bool
+
+	// SendOwedStfu sends Stfu if it owes one. It returns an error if the
+	// state machine is in an invalid state.
+	SendOwedStfu(numPendingLocalUpdates uint64) error
```

Comment:
re the name `numPendingLocalUpdates` - I take it as the number of pending updates from the local logs compared to both the remote commitment and the local commitment, is that correct?

Reply:
This is correct. Examining this again though I'm realizing that your desire to just do this check in the outer scope makes sense. The number we feed in here is actually inconsequential. The only thing that matters here is whether this number is zero or not. We could just as easily feed in a boolean that is `hasLocalUpdates bool`. At that point it just makes more sense to check it in the outer scope.

Reply:
Most recent commits once again remove this. This demonstrates the before and after. I think the tradeoff is worth it.